integration_tests/CMakeLists.txt

```diff
@@ -427,4 +427,5 @@ RUN(NAME comp_01 LABELS cpython llvm wasm wasm_x64)
 RUN(NAME bit_operations_i32 LABELS cpython llvm wasm wasm_x64)
 RUN(NAME bit_operations_i64 LABELS cpython llvm wasm)
 
-RUN(NAME test_argv_01    LABELS llvm) # TODO: Test using CPython
+RUN(NAME test_argv_01      LABELS llvm) # TODO: Test using CPython
+RUN(NAME global_syms_01    LABELS cpython)
```

Comment:
Why not `llvm` and `c` here?

Reply:
In LLVM, the global lists are not yet handled. After merging this I will start working on it.
C: I didn't try it. How do I verify this works in C?
Using `lpython integration_tests/global_syms_01.py --show-c`?

Reply:
> In LLVM, the global lists are not yet handled. After merging this I will start working on it.

Any rough idea on how that will be managed. I think LLVM doesn't allow `builder->CreateAlloca` in global scope. Worth researching approaches (implementation in LLVM is not at all required just a rough idea that there is a possibility to make it work in LLVM) once this   before merging this because the success of this might rely on the possibility of actually using it in LLVM.

One approach in my mind is to create `void*` typed global variables in the global scope and then bitcast them into `list` LLVM type inside LLVM functions. That way any manipulation will happen inside LLVM functions.

Other approach is to see how clang++ does it for global variables like `std::vector` in C++. For example the below code,

```cpp
#include <iostream>
#include <vector>

std::vector<int> vec;

int main() {
    return 0;
}
```

`clang++`'s LLVM output,

```llvm
; ModuleID = '/Users/czgdp1807/lpython_project/debug.cpp'
source_filename = "/Users/czgdp1807/lpython_project/debug.cpp"
target datalayout = "e-m:o-i64:64-i128:128-n32:64-S128"
target triple = "arm64-apple-macosx13.0.0"

%"class.std::__1::vector" = type { %"class.std::__1::__vector_base" }
%"class.std::__1::__vector_base" = type { i32*, i32*, %"class.std::__1::__compressed_pair" }
%"class.std::__1::__compressed_pair" = type { %"struct.std::__1::__compressed_pair_elem" }
%"struct.std::__1::__compressed_pair_elem" = type { i32* }
%"struct.std::__1::nullptr_t" = type { i8* }
%"struct.std::__1::__default_init_tag" = type { i8 }
%"class.std::__1::__vector_base_common" = type { i8 }
%"struct.std::__1::__compressed_pair_elem.0" = type { i8 }
%"class.std::__1::allocator" = type { i8 }
%"struct.std::__1::__non_trivial_if" = type { i8 }

@vec = global %"class.std::__1::vector" zeroinitializer, align 8
@__dso_handle = external hidden global i8
@llvm.global_ctors = appending global [1 x { i32, void ()*, i8* }] [{ i32, void ()*, i8* } { i32 65535, void ()* @_GLOBAL__sub_I_debug.cpp, i8* null }]

; Function Attrs: noinline ssp uwtable
define internal void @__cxx_global_var_init() #0 section "__TEXT,__StaticInit,regular,pure_instructions" {
  %1 = call %"class.std::__1::vector"* @_ZNSt3__16vectorIiNS_9allocatorIiEEEC1Ev(%"class.std::__1::vector"* @vec)
  %2 = call i32 @__cxa_atexit(void (i8*)* bitcast (%"class.std::__1::vector"* (%"class.std::__1::vector"*)* @_ZNSt3__16vectorIiNS_9allocatorIiEEED1Ev to void (i8*)*), i8* bitcast (%"class.std::__1::vector"* @vec to i8*), i8* @__dso_handle) #2
  ret void
}
```

It can be seen that it creates global variables but initialises them inside a function. So you can do something similar for lists as well. Also, worth noting that we will have to create a function something like `global_module_function` in LLVM and then call inside `main`. But that should be doable.

> C: I didn't try it. How do I verify this works in C?
> Using `lpython integration_tests/global_syms_01.py --show-c`?

If LLVM is not enabled then C is not worth enabling. LLVM should always work with a given ASR design then other backends will automatically follow.

Reply:
I am approving it as I think this is clean enough at the ASR as well as LLVM/C level.

Reply:
@czgdp1807 the question here is how to handle module variables, as well as program variables. They are scoped to the module or program, but are essentially global. WASM and x86 have the same issue (@Shaikh-Ubaid and I discussed this a few times). It seems one must allocate the global variable (using the platform dependent mechanism), and then initialize it from the main function as needed.

Reply:
So seems like global statements are already being wrapped into a main function in the ASR itself? Well I would like to see what we will need to do when we enable `llvm` for this test. Let’s merge it for now (I have approved earlier today).

Reply:
Let me review this first.